src/main/scala/org/broadinstitute/monster/etl/encode/ExtractionPipeline.scala

```diff
@@ -0,0 +1,72 @@
+package org.broadinstitute.monster.etl.encode
+
+import caseapp.{AppName, AppVersion, HelpMessage, ProgName}
+import com.spotify.scio.ContextAndArgs
+import com.spotify.scio.extra.json._
+import org.broadinstitute.monster.EncodeIngestBuildInfo
+
+/**
+  * ETL workflow for scraping the latest entity metadata from ENCODE.
+  */
+object ExtractionPipeline {
+
+  @AppName("ENCODE Extraction Pipeline")
+  @AppVersion(EncodeIngestBuildInfo.version)
+  @ProgName("org.broadinstitute.monster.etl.encode.ExtractionPipeline")
+  /**
+    * Command-line arguments for the ETL workflow.
+    *
+    * scio's `ContextAndArgs.typed` delegates down to `caseapp`, which will generate
+    * parsers + help text for these args (as well as Beams' underlying options)
+    */
+  case class Args(
+    @HelpMessage(
+      "Path to directory where the extracted raw ENCODE metadata should be written"
+    )
+    outputDir: String
+  )
+
+  /**
+    * pulling raw metadata using the ENCODE search client API for the following specific entity types...
+    * Biosamples, donors, and libraries
+    */
+  def main(rawArgs: Array[String]): Unit = {
+    val (pipelineContext, parsedArgs) = ContextAndArgs.typed[Args](rawArgs)
+
+    val biosamples = {
+      val additionalParams =
+        pipelineContext.parallelize(List(List("organism.name" -> "human")))
+
+      EncodeExtractions.getEntities(EncodeEntity.Biosample)(additionalParams)
+    }
```

Comment:
What do you think of this block pattern? I'm not sure I like it, but I could be convinced if others find it useful.

Reply:
I think it's nice and tidy? Would the alternative be to just not have them in blocks and like assign the output of line 40 to `val biosamples`?

Reply:
We could use the curry-ing built into the `getEntities` signature to inline things, like this:
```scala
val biosamples = EncodeExtractions.getEntities(EncodeEntity.Biosample) {
  pipelineContext.parallelize(List(List("organism.name" -> "human")))
}
```
I'm also wondering if there's enough repetition here to make a generic method that we call over and over again. I imagine that method would also include the `saveAsJsonFile` piece.